pkg/util/servicediscovery/ring.go

```diff
@@ -53,43 +59,65 @@ func (r *ringServiceDiscovery) running(ctx context.Context) error {
 	defer ringTicker.Stop()
 
 	// Notifies the initial state.
-	ringState, _ := r.ringClient.GetAllHealthy(r.ringOp) // nolint:errcheck
-	r.notifyChanges(ringState)
+	all, _ := r.ringClient.GetAllHealthy(activeRingOp) // nolint:errcheck
+	r.notifyChanges(all)
 
 	for {
 		select {
 		case <-ringTicker.C:
-			ringState, _ := r.ringClient.GetAllHealthy(r.ringOp) // nolint:errcheck
-			r.notifyChanges(ringState)
+			all, _ := r.ringClient.GetAllHealthy(activeRingOp) // nolint:errcheck
+			r.notifyChanges(all)
 		case <-ctx.Done():
 			return nil
 		case err := <-r.subservicesWatcher.Chan():
-			return errors.Wrap(err, "a subservice of ring-based service discovery has failed")
+			return errors.Wrap(err, "a subservice of query-schedulers ring-based service discovery has failed")
```

Comment:
This is a generic package, so why does it mention "query-scheduler" in the error message? Component using ring-based discovery can add context to the error message if it wants.

Reply:
It's typo. Thanks for spotting it!